lib/Slack_web_api.js

```diff
@@ -1,314 +1,180 @@
-            },
-            history: function(options, cb) {
-                slack_api.callAPI('channels.history', options, cb);
-            },
-            info: function(options, cb) {
-                slack_api.callAPI('channels.info', options, cb);
-            },
-            invite: function(options, cb) {
-                slack_api.callAPI('channels.invite', options, cb);
-            },
-            join: function(options, cb) {
-                slack_api.callAPI('channels.join', options, cb);
-            },
-            kick: function(options, cb) {
-                slack_api.callAPI('channels.kick', options, cb);
-            },
-            leave: function(options, cb) {
-                slack_api.callAPI('channels.leave', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('channels.list', options, cb);
-            },
-            mark: function(options, cb) {
-                slack_api.callAPI('channels.mark', options, cb);
-            },
-            rename: function(options, cb) {
-                slack_api.callAPI('channels.rename', options, cb);
-            },
-            setPurpose: function(options, cb) {
-                slack_api.callAPI('channels.setPurpose', options, cb);
-            },
-            setTopic: function(options, cb) {
-                slack_api.callAPI('channels.setTopic', options, cb);
-            },
-            unarchive: function(options, cb) {
-                slack_api.callAPI('channels.unarchive', options, cb);
+
+                return cb(json.error, json);
             }
-        },
-        chat: {
-            delete: function(options, cb) {
-                slack_api.callAPI('chat.delete', options, cb);
-            },
-            postMessage: function(options, cb) {
-                if (options.attachments && typeof(options.attachments) != 'string') {
-                    options.attachments = JSON.stringify(options.attachments);
+
+            return cb(error || 'Invalid response');
+        }).form(options);
+    };
+
+    slack_api.callAPIWithoutToken = function(command, options, cb) {
+        cb = cb || noop;
+
+        bot.log('** API CALL: ' + slack_api.api_url + command);
+        if (!options.client_id) {
+            options.client_id = bot.config.clientId;
+        }
+        if (!options.client_secret) {
+            options.client_secret = bot.config.clientSecret;
+        }
+        if (!options.redirect_uri) {
+            options.redirect_uri = bot.config.redirectUri;
+        }
+        request.post(slack_api.api_url + command, function(error, response, body) {
+            bot.debug('Got response', error, body);
+            if (!error && response.statusCode == 200) {
+                var json;
+                try {
+                    json = JSON.parse(body);
+                } catch (err) {
+                    return cb(err || 'Invalid JSON');
```

Comment:
`err` is an instance of `Error` class but if `err` doesn't exist, and the body is an invalid JSON you're sending `String`, therefore it makes the compatibility much harder to comprehend.